edgelet/kube-client/src/client.rs

```diff
@@ -343,6 +357,7 @@ where
     fn request<R: K8sResponse>(
         &mut self,
         req: http::Request<Vec<u8>>,
+        _response_body: fn(http::StatusCode) -> ResponseBody<R>,
```

Comment:
As discussed offline, you *can* do this, but it would be more convenient for the callers to take a tuple here `(req, _response_body): (http::StatusCode, fn (http::StatusCode) -> ResponseBody<R>)` so that they can do `|req| self.request(req)` instead of `|(req, response_body)| self.request(req, response_body)`.

If the clippy `type_complexity` lint is the problem it should be sufficient to `#[allow]` it on just the `Client::request` definition.

Reply:
Done